cmd/curio/guidedsetup/guidedsetup.go

```diff
@@ -0,0 +1,593 @@
+			os.Exit(1)
+		}
+		_, err = (&promptui.Prompt{
+			Label: d.T("Press return to update %s with Yugabyte info. Backup the file now.", "config.toml")}).Run()
+		if err != nil {
+			os.Exit(1)
+		}
+		p, err := homedir.Expand(d.MinerConfigPath)
+		if err != nil {
+			d.say(notice, "Error expanding path: %s", err.Error())
+			os.Exit(1)
+		}
+		stat, err := os.Stat(path.Join(p, "config.toml"))
+		if err != nil {
+			d.say(notice, "Error reading filemode of config.toml: %s", err.Error())
+			os.Exit(1)
+		}
+
+		filemode := stat.Mode()
+		err = os.WriteFile(path.Join(p, "config.toml"), buf, filemode)
+		if err != nil {
+			d.say(notice, "Error writing config.toml: %s", err.Error())
+			os.Exit(1)
+		}
+		d.say(section, "Restart Lotus Miner. ")
+	}
+	stepCompleted(d, d.T("Connected to Yugabyte"))
+}
+
+func readMinerConfig(d *MigrationData) {
+	d.say(plain, "To start, ensure your sealing pipeline is drained and shut-down lotus-miner.")
+
+	verifyPath := func(dir string) (*config.StorageMiner, error) {
+		cfg := config.DefaultStorageMiner()
+		dir, err := homedir.Expand(dir)
+		if err != nil {
+			return nil, err
+		}
+		_, err = toml.DecodeFile(path.Join(dir, "config.toml"), &cfg)
+		return cfg, err
+	}
+
+	dirs := map[string]*config.StorageMiner{"~/.lotusminer": nil, "~/.lotus-miner-local-net": nil}
+	if v := os.Getenv("LOTUS_MINER_PATH"); v != "" {
+		dirs[v] = nil
+	}
+	for dir := range dirs {
+		cfg, err := verifyPath(dir)
+		if err != nil {
+			delete(dirs, dir)
+		}
+		dirs[dir] = cfg
+	}
+
+	var otherPath bool
+	if len(dirs) > 0 {
+		_, str, err := (&promptui.Select{
+			Label:     d.T("Select the location of your lotus-miner config directory?"),
+			Items:     append(lo.Keys(dirs), d.T("Other")),
+			Templates: d.selectTemplates,
+		}).Run()
+		if err != nil {
+			if err.Error() == "^C" {
+				os.Exit(1)
+			}
+			otherPath = true
+		} else {
+			if str == d.T("Other") {
+				otherPath = true
+			} else {
+				d.MinerConfigPath = str
+				d.MinerConfig = dirs[str]
+			}
+		}
+	}
+	if otherPath {
+	minerPathEntry:
+		str, err := (&promptui.Prompt{
+			Label: d.T("Enter the path to the configuration directory used by %s", "lotus-miner"),
+		}).Run()
+		if err != nil {
+			d.say(notice, "No path provided, abandoning migration ")
+			os.Exit(1)
+		}
+		cfg, err := verifyPath(str)
+		if err != nil {
+			d.say(notice, "Cannot read the config.toml file in the provided directory, Error: %s", err.Error())
+			goto minerPathEntry
+		}
+		d.MinerConfigPath = str
+		d.MinerConfig = cfg
+	}
+
+	// Try to lock Miner repo to verify that lotus-miner is not running
+	{
+		r, err := repo.NewFS(d.MinerConfigPath)
+		if err != nil {
+			d.say(plain, "Could not create repo from directory: %s. Aborting migration", err.Error())
+			os.Exit(1)
+		}
+		lr, err := r.Lock(repo.StorageMiner)
+		if err != nil {
+			d.say(plain, "Could not lock miner repo. Your miner must be stopped: %s\n Aborting migration", err.Error())
+			os.Exit(1)
+		}
+		_ = lr.Close()
```

Comment:
Do we not want to hold this lock during the config-changing steps? Until `lr.Close` is called it will not be possible to start lotus-miner, which in this case is probably what we want

Reply:
There's no reason for users to be "devious" here, and we get new locks in that part of the code.
For that to be a problem, the would have started it during Yugabyte config or during the config-write step. 
I think we should allow them to run it. They may be blindsided about needing Yugabyte and start lotus-miner while they figure that step out. 
Thoughts @magik6k ?

Reply:
Yeah, let's keep this simple for now, we'll see how users manage to break this in the real world anyways, and it will probably be things we didn't even imagine while probably not being this